src/function/table/show_macros.cpp

```diff
@@ -0,0 +1,101 @@
+#include "binder/binder.h"
+#include "catalog/catalog.h"
+#include "catalog/catalog_entry/scalar_macro_catalog_entry.h"
+#include "catalog/catalog_entry/table_catalog_entry.h"
+#include "function/table/bind_data.h"
+#include "function/table/simple_table_function.h"
+#include "main/client_context.h"
+#include "main/database_manager.h"
+
+using namespace kuzu::common;
+using namespace kuzu::catalog;
+
+namespace kuzu {
+namespace function {
+
+struct MacroInfo {
+    std::string name;
+    std::string definition;
+
+    MacroInfo(std::string name, std::string definition)
+        : name{std::move(name)}, definition(std::move(definition)) {}
+};
+
+struct ShowMacrosBindData final : TableFuncBindData {
+    std::vector<MacroInfo> macros;
+
+    ShowMacrosBindData(std::vector<MacroInfo> macros, binder::expression_vector columns,
+        row_idx_t numRows)
+        : TableFuncBindData{std::move(columns), numRows}, macros{std::move(macros)} {}
+
+    std::unique_ptr<TableFuncBindData> copy() const override {
+        return std::make_unique<ShowMacrosBindData>(macros, columns, numRows);
+    }
+};
+
+static offset_t internalTableFunc(const TableFuncMorsel& morsel, const TableFuncInput& input,
+    DataChunk& output) {
+    const auto macros = input.bindData->constPtrCast<ShowMacrosBindData>()->macros;
+    const auto numMacrosToOutput = morsel.endOffset - morsel.startOffset;
+    for (auto i = 0u; i < numMacrosToOutput; i++) {
+        const auto tableInfo = macros[morsel.startOffset + i];
+        output.getValueVectorMutable(0).setValue(i, tableInfo.name);
+        output.getValueVectorMutable(1).setValue(i, tableInfo.definition);
+    }
+    return numMacrosToOutput;
+}
+
+static std::unique_ptr<TableFuncBindData> bindFunc(const main::ClientContext* context,
+    const TableFuncBindInput* input) {
+    std::vector<std::string> columnNames;
+    std::vector<LogicalType> columnTypes;
+    columnNames.emplace_back("name");
+    columnTypes.emplace_back(LogicalType::STRING());
+    columnNames.emplace_back("definition");
+    columnTypes.emplace_back(LogicalType::STRING());
+    std::vector<MacroInfo> macroInfos;
+    auto transaction = context->getTransaction();
+    if (!context->hasDefaultDatabase()) {
```

Comment:
@acquamarin should comment this. I feel the current design will not scale if we keep adding functions that needs to check default database and attached database separately. The ideal case should be getting the current database and check its macro. 

Also, do have ways to test attached database macros?